### ISSUE TITLE
zodios-react breaks react-query v4 notifyOnChangeProps default tracked behavior

### ISSUE THREAD

**Issue by QzCurious** [ecyrbe/zodios#264]
## How react-query works

I don't know how TkDodo makes react-query [smartly track the props for a `useQuery`](https://tanstack.com/query/v4/docs/guides/migrating-to-react-query-4#notifyonchangeprops-property-no-longer-accepts-tracked-as-a-value), maybe by `Proxy`.
For the following code, `useQuery` can now smartly re-render itself only once if `isError` is changed. (would not re-render if `data` prop is changed)

```ts
const { isError } = useQuery(
  ['any-key'],
  async () => { return true }
)
```

The fact seems to be that `useQuery` would only re-render for a prop if it is **accessed**.

## The Issue

zodios-react breaks the behavior by [destructing `useQuery`](https://github.com/ecyrbe/zodios-react/blob/39481de198e3b87042daa8254f53f8ed691eee67/src/hooks.ts#L235).

```ts
return {
  invalidate,
  key,
  ...useQuery(key, query, queryOptions),
} as UseQueryResult<TData, Errors> & {
  invalidate: () => Promise<void>;
  key: QueryKey;
};
```

A quick way to solve this might be like (not try it yet):

```ts
const query = useQuery(key, query, queryOptions) as UseQueryResult<TData, Errors> & {
  invalidate: () => Promise<void>;
  key: QueryKey;
}
query.invalidate = invalidate
query.key = key
return query
```

**Comment by ecyrbe** [comment]
Yes this should work. That's actually how it's implemented for solid version

**Comment by ecyrbe** [comment]
This is now fixed in `v10.4.2` of `@zodios/react`
